src/server/node.rs

```diff
@@ -192,16 +192,43 @@ impl<C> Node<C>
               peer_id,
               region_id);
 
-        let region = try!(store::bootstrap_region(engine, store_id, region_id, peer_id));
+        let region = try!(store::prepare_bootstrap(engine, store_id, region_id, peer_id));
         Ok(region)
     }
 
+    fn check_prepare_bootstrap_cluster(&self, engine: &DB) -> Result<()> {
+        let res = try!(engine.get_msg::<metapb::Region>(&keys::prepare_bootstrap_key()));
+        if res.is_none() {
+            return Ok(());
+        }
+
+        let region_id = res.unwrap().get_id();
+        for _ in 0..MAX_CHECK_CLUSTER_BOOTSTRAPPED_RETRY_COUNT {
+            match self.pd_client.get_region(b"") {
+                Ok(region) => {
+                    if region.get_id() == region_id {
```

Comment:
I think we must check region epoch, if not equal, there must be some fatal error here.